R/extremes-data.R

```diff
@@ -68,92 +94,50 @@ extremesTable <- function(rawData){
+          data.frame(name=rowName, date=dateTime[,1], time=timeFormatting, primary=primaryValue, related=relatedValue, stringsAsFactors = FALSE)
+        } else {
+          data.frame(name=rowName, date=dateTime[,1], time=timeFormatting, primary=relatedValue, related=primaryValue, stringsAsFactors = FALSE)
         }
-        
       } else {
-        if (is.null(x$points$value)) {
-          primary <- "N/A"
-        }
-        else {
-          primary <- x$points$value  
-        }
-        upchain  <- "N/A"
+        data.frame(name=rowName, date=dateTime[,1], time=timeFormatting, primary=primaryValue, stringsAsFactors = FALSE)
       }
-      
-      data.frame(dateTime[,1], timeFormatting, primary, upchain, stringsAsFactors = FALSE)
-      
-      
     })
-    
-    names(min.max) <- paste0("data$", names(data)[i], "$", names(subset))
-    results <- append(results, min.max) 
-    
-  }
-  
-  results <- orderMaxMin(results, data$reportMetadata$isInverted)
-  
-  #Change column and row names to their correct forms and add them into the dataframe.
-  toRet <- data.frame()
-  for(i in 1:length(results)){
-    toAdd <- cbind(c(orderedRowNames[i],rep("",nrow(results[[i]])-1)),results[[i]]) 
-    colnames(toAdd) <- columnNames
-    rownames(toAdd) <- NULL
-    if (nrow(toAdd)>1) {
-      temp <- toAdd
-      upchain <- paste("Upchain series ", upchainParameter, " (", upchainUnit, ")")
-      primary <- paste("Primary series ", primaryParameter, " (", primaryUnit, ")")
-      colnames(temp) <- c("Temp", "Date", "Time", primary, upchain)
-      colnames(toAdd) <- c("Temp", "Date", "Time", primary, upchain)
-      if (grepl("Min", orderedRowNames[i])) {
-        param <- "min"
-      } else {
-        param <- "max"
+
+    #Select Proper Data
+    if(!is.null(dataRows[[1]])){
+      dataRows <- dataRows[[1]]
+
+      #Remove Unnecessary Data
+      if(includeRelated && !isUpchain && doMerge){
+        dataRows <- dataRows[!duplicated(dataRows[c("date", "related")]),]
+      } else if(doMerge) {
+        dataRows <- dataRows[!duplicated(dataRows[c("date", "primary")]),]
       }
-      oneDaily <- aggregate(temp[[5]] ~ temp[[2]], temp, param)
-      colnames(oneDaily) <- c("Date", upchain)
-      merged <- merge(oneDaily, toAdd, by=c("Date", upchain), all.x=TRUE)
-      merged <- merged[!duplicated(merged[,c('Date', upchain)]),]
-      colnames(merged) <- c("Date", upchain, "Temp", "Time", primary)
-      merged <- merged[c("Temp", "Date", "Time", primary, upchain)]
-      merged$Date <- as.Date(merged$Date,format = "%m-%d-%Y")
-      merged <- merged[order(merged$Date), ]
-      merged$Temp[1] <- c(orderedRowNames[i])
-      merged$Date <- as.character(merged$Date, format = "%m-%d-%Y")
-      toAdd <- merged
-      colnames(toAdd) <- columnNames
-    } 
-    toRet <- rbind(toRet,toAdd)
-    
-  }
-  
-  return(toRet)
+
+      #Replace Duplicate Names with blank names
+      if(NROW(dataRows[duplicated(dataRows["name"]),]) > 0){
```

Comment:
hmm I've never seen that function capitalized before -- but it does the same thing as `nrow` so :+1: